sfcasts/ep3-doctrine/rich-model.md

```diff
@@ -1,74 +1,21 @@
-more descriptive, like a vote and down vote. All right, next, let's install an
-awesome library called doctrine extensions. This is like a magic library full of
-superpowers, like automatic timetable in slug creation behaviors.
+All right, look back at `MixController.php`. The logic for doing our "up" and "down" voting is pretty simple, but I think it could be more clear. Try this! Open up `VinylMix.php` and scroll down to find `setVotes`. Right after this, just to keep things organized, create a new `public function` called `upVote()` and return `void`. Inside, say `$this->votes++`, copy that, and create a *second* method which we'll call - you guessed it - `downVote()` with `$this->votes--`. Thanks to these methods, in `MixController.php`, instead of having `$mix->setVotes()` set to `$mix getVotes() + 1`, we can just say `$mix->upVote()` and `$mix->downVote()`. Now *that's* nice. Our controller reads much more clearly, and we've encapsulated the `upVote()` and `downVote()` logic into the entity. If we head over and refresh, it *still* works.
 
+Okay, this highlights an important topic. We've added *three* custom methods to our entity. One that helps read the data in a special way, and two that help *set* the data. When we run `make:entity`, it creates getter and setter methods for every single property. That's nice, because it makes our entity *infinitely* flexible. *Anyone* from *anywhere* can read or set any property. But sometimes, you might *not* want or need that. For example, do we really want a `setVotes()` method? Is there really a use case in our code for something to set the vote count to *any* number that it wants? Probably not. We'll likely only need `upVote()` and `downVote()`. I will keep the `setVotes()` method for now, though, because we'll use it when we generate our dummy vinyl mix.
+
+But this touches on a really interesting idea. By removing unnecessary getter and setter methods in your entity and replacing them with more descriptive methods like `upVote()`, `downVote()`, `getVoteString()`, or `getImageUrl()` - methods that fit your business logic - you can, little by little, give your entities more clarity. Our `upVote()` and `downVote()` methods are super clear and descriptive. Someone calling these doesn't even need to know or *care* how they work internally.
+
+Entities that *only* have getter and setter methods are sometimes called "anemic models". Entities that *remove* these and replace them with specific methods for your business logic are sometimes called "smart models". Some people take this to an extreme and have almost *no* getter or setter methods. Here at SymfonyCasts, we tend to be more pragmatic. We usually do have getter and setter methods, but we always look for ways to be more descriptive, like `upVote()` and `downVote()`.
+
+Next, let's install an *awesome* library called DoctrineExtensions. This is a magic library full of superpowers, like automatic, timestampable, and slug creation behaviors.
```

Comment:
> ... And now I'm wondering if "timestampable" should read "timestamp-able"...
https://github.com/doctrine-extensions/DoctrineExtensions#orm--mongodb-odm
Good question! We write "timestampable" as a single word w/o a dash, that's the name of many behaviors from that bundle. The same rule applies for other behaviors like "slugable", "translatable", etc. For the reference: https://github.com/doctrine-extensions/DoctrineExtensions#orm--mongodb-odm


And so, I think we can drop those comas from the last sentence:
```suggestion
Next, let's install an *awesome* library called DoctrineExtensions. This is a magic library full of superpowers, like automatic timestampable and slug creation behaviors.
```